File: 0003.Longest_Substring_Without_Repeating_Characters/Solution.php
<?php

class Solution {

    /**
     * @param String $s
     * @return Integer
     */
    function lengthOfLongestSubstring($s) {
        $dict = array();
        $max = $preLen = $curLen = 0;

        for ($i = 0; $i < strlen($s); $i++) {
            if (!isset($dict[$s[$i]])) {
                $curLen = $preLen + 1;
            } else {
                $conLen = $preLen + 1;
                $posLen = $i - $dict[$s[$i]];
                $curLen = $conLen < $posLen ? $conLen : $posLen;
            }

            $max = $max < $curLen ? $curLen : $max;
            $dict[$s[$i]] = $i;
            $preLen = $curLen;
        }

        return $max;
    }
}

File: 0003.Longest_Substring_Without_Repeating_Characters/solution.go
package solution

func lengthOfLongestSubstring(s string) int {
	dict := map[byte]int{}
	max, preLen, curLen := 0, 0, 0

	for i := 0; i < len(s); i++ {
		pos, ok := dict[s[i]]

		if !ok {
			curLen = preLen + 1
		} else {
			conLen := preLen + 1
			posLen := i - pos

			if conLen < posLen {
				curLen = conLen
			} else {
				curLen = posLen
			}
		}

		if max < curLen {
			max = curLen
		}

		dict[s[i]] = i
		preLen = curLen
	}

	return max
}
